src/main/java/com/mathworks/ci/TestFile.java

```diff
@@ -0,0 +1,94 @@
+package com.mathworks.ci;
+
+/**
+ * Copyright 2024, The MathWorks Inc.
+ *
+ */
+
+import java.util.List;
+import java.util.ArrayList;
+
+import org.apache.commons.lang.RandomStringUtils;
+
+public class TestFile {
+    private String filePath;
+    private String name;
+    private Double duration;
+    private String status;
+    private List<TestCase> testCases;
+    private String id;
+
+    public TestFile() {
+        filePath = "";
+        name = "";
+        duration = 0.0;
+        status = "NotRun";
+        testCases = new ArrayList<TestCase>();
+
+        id = RandomStringUtils.randomAlphanumeric(8);
+    }
+
+    public void incrementDuration(Double testCaseDuration) {
+        this.duration += testCaseDuration;
+    }
+
+    public void updateStatus(TestCase testCase) {
+        if (!status.equals("Failed")){

```

Comment:
Although its hardcoded its good to check it with equalsIgnoreCase() for future changes from Failed to failed

Reply:
Using static constants now